src/core/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,244 @@
+   */
+  id?: string;
+  /** Callback fired when button is clicked */
+  onClick?: () => void;
+  /** Callback fired on button onBlur
+   * @param {FocusEvent<HTMLButtonElement>} event FocusEvent
+   */
+  onBlur?: (event: FocusEvent<HTMLButtonElement>) => void;
+  /** Visually hidden label text for the button */
+  openButtonLabel: string;
+  /** Callback fired when menu opens */
+  onClose?: () => void;
+  /** Callback fired when menu closes */
+  onOpen?: () => void;
+  /**
+   * Props which are placed at the outermost div of the component.
+   * Can be used, for example, for style
+   */
+  wrapperProps?: Omit<
+    HTMLAttributesIncludingDataAttributes<HTMLDivElement>,
+    'className'
+  >;
+}
+
+const BaseActionMenu = (props: ActionMenuProps) => {
+  const {
+    className,
+    wrapperProps,
+    id,
+    fullWidth,
+    forwardedRef,
+    buttonText,
+    openButtonLabel,
+    children,
+    onOpen,
+    onClose,
+    onClick,
+    onBlur,
+    ...passProps
+  } = props;
+
+  const openButtonRef = forwardedRef || useRef<HTMLButtonElement>(null);
+  const [menuVisible, setMenuVisible] = useState<boolean>(false);
+  const [selectFirstItem, setSelectFirstItem] =
+    useState<InitialActiveDescendant>('none');
+
+  const menuId = `${id}-menu`;
+  const buttonId = `${id}-button`;
+
+  const openMenu = () => {
+    if (onOpen) {
+      onOpen();
+    }
+    // Highlighting the first item is a compomise to keep NVDA smooth
+    setSelectFirstItem('first');
+    setMenuVisible(true);
+  };
+
+  const closeMenu = () => {
+    setMenuVisible(false);
+    if (onClose) {
+      onClose();
+    }
+
+    // Move focus back to the button when menu is closed
+    openButtonRef.current?.focus();
+  };
+
+  const handleButtonClick = () => {
+    if (onClick) {
+      onClick();
+    }
+
+    if (menuVisible) {
+      closeMenu();
+    } else {
+      openMenu();
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'ArrowUp' && !menuVisible) {
+      event.preventDefault();
+      openMenu();
+      setSelectFirstItem('last');
+    }
+
+    if (
+      (event.key === 'ArrowDown' ||
+        event.key === 'Enter' ||
+        event.key === ' ') &&
+      !menuVisible
+    ) {
+      event.preventDefault();
+      openMenu();
+      setSelectFirstItem('first');
+    }
+  };
+
+  return (
+    <HtmlDiv
+      {...wrapperProps}
+      className={classnames(baseClassName, className, {
+        [actionMenuClassNames.disabled]: !!passProps.disabled,
+        [actionMenuClassNames.fullWidth]: fullWidth,
+      })}
+    >
+      <>
+        <Button
+          id={buttonId}
+          variant="secondary"
+          iconRight="optionsVertical"
+          aria-expanded={menuVisible}
+          aria-controls={menuId}
+          forwardedRef={openButtonRef}
+          className={classnames(
+            actionMenuClassNames.button,
+            {
+              [actionMenuClassNames.buttonDisabled]: passProps.disabled,
```

Comment:
Other classnames can be added to this same object instead of multiple objects

Reply:
Fixed